.github/workflows/ci.yml

```diff
@@ -87,6 +87,56 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true' && matrix.cache == 'true'
         run: rm -rf ~/.m2/repository/io/trino/trino-*
 
+  reproducible-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+          ref: |
+            ${{ github.event_name == 'repository_dispatch' &&
+                github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha &&
+                format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
+      - uses: ./.github/actions/setup
+        with:
+          cache: 'false'
+          cleanup-node: true
+      - name: First build to a local repository
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} -Dmaven.repo.local=build_1
+      - name: Second build to a local repository
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} -Dmaven.repo.local=build_2
+      - name: Compare builds
+        run: |
+          diff \
+              --exclude maven-metadata-local.xml \
+              --exclude=_remote.repositories \
+              --exclude="*.lastUpdated" \
+              --exclude="resolver-status.properties" \
+              -bur build_1/io/trino build_2/io/trino > /dev/null 2>&1
+          error=$?
+          if [ $error -eq 0 ]
+          then
+            echo "First and second build produced the same artifacts"
+            exit 0
+          elif [ $error -eq 1 ]
+          then
+            echo "First and second build are not the same"
+            diff \
+                --exclude maven-metadata-local.xml \
+                --exclude=_remote.repositories \
+                --exclude="*.lastUpdated" \
+                --exclude="resolver-status.properties" \
+                -bur build_1/io/trino build_2/io/trino
```

Comment:
This non-incidentally looks very similar to the above, except for `> /dev/null 2>&1`

why not run the `diff` only once and rely on `set -e` to produce CI job failure when diff fails?
the `diff` invocations would't be able to drift, if we had only one